java/client/src/org/openqa/selenium/support/ui/ExpectedConditions.java

```diff
@@ -933,4 +934,613 @@ private static WebElement findElement(By by, WebDriver driver) {
+          } catch (Exception e) {/**/}
+        }
+        return Boolean.FALSE;
+      }
+
+      @Override
+      public String toString() {
+        String message = "at least one condition to be valid: ";
+        for (ExpectedCondition<?> condition : conditions) {
+          message += condition.toString();
+        }
+        return message;
+      }
+    };
+  }
+
+
+  /**
+   * An expectation with the logical and condition of the given list of conditions.
+   *
+   * Each condition is checked until all of them return true or not null
+   *
+   * @param conditions ExpectedCondition is a list of alternative conditions
+   * @return true once all conditions are satisfied
+   */
+  public static ExpectedCondition<Boolean> and(final ExpectedCondition<?>... conditions) {
+    return new ExpectedCondition<Boolean>() {
+      @Override
+      public Boolean apply(WebDriver driver) {
+        for (ExpectedCondition<?> condition : conditions) {
+          try {
+            Object result = condition.apply(driver);
+            if (result == null) {
+              return Boolean.FALSE;
+            } else if (result instanceof Boolean) {
+              if (!(Boolean) result) {
+                return Boolean.FALSE;
+              }
+            }
+          } catch (Exception e) {
+            return Boolean.FALSE;
+          }
+        }
+        return Boolean.TRUE;
+      }
+
+      @Override
+      public String toString() {
+        String message = "all conditions to be valid: ";
+        for (ExpectedCondition<?> condition : conditions) {
+          message += condition.toString();
+        }
+        return message;
+      }
+    };
+  }
+
+
+  /**
+   * An expectation to check if js executable
+   *
+   * Usefull when  you know, that there should be js val or something at the stage
+   *
+   * @param javaScript used as executable script
+   * @return true once javaScript executed without errors
+   */
+  public static ExpectedCondition<Boolean> jsReturnsNoErrors(
+    final String javaScript) {
+    return new ExpectedCondition<Boolean>() {
+      @Override
+      public Boolean apply(WebDriver driver) {
+        try {
+          ((JavascriptExecutor) driver).executeScript(javaScript);
+          return Boolean.TRUE;
+        } catch (Exception e) {
+          return Boolean.FALSE;
+        }
+      }
+
+      @Override
+      public String toString() {
+        return String.format("js %s to be executable", javaScript);
+      }
+    };
+  }
+
+  /**
+   * An expectation for String value from javascript
+   *
+   * @param javaScript as executable js line
+   * @return true once js return string
+   */
+  public static ExpectedCondition<String> jsReturnsValue (
+    final String javaScript) {
+    return new ExpectedCondition<String>() {
+      @Override
+      public String apply(WebDriver driver) {
+        String value = null;
+        try {
+          value = (String) ((JavascriptExecutor) driver).executeScript(javaScript);
```

Comment:
Definitely shouldn't assume the returned object is a String. The javascript executor can return a number of Objects see https://github.com/SeleniumHQ/selenium/blob/master/java/client/src/org/openqa/selenium/remote/internal/JsonToWebElementConverter.java

Would be better to assume Object and check that it isn't null, then check if it is an instance of string and that it's not empty or if it's an instance of List and also not empty